src/utils/mouse-events-handler.js

```diff
@@ -1,28 +1,34 @@
-const HORIZONTAL_LOOK_SPEED = 0.0035;
-const VERTICAL_LOOK_SPEED = 0.0021;
-const PI_4 = Math.PI / 4;
+// TODO: Make look speed adjustable by the user
+const HORIZONTAL_LOOK_SPEED = 0.1;
+const VERTICAL_LOOK_SPEED = 0.06;
 
 export default class MouseEventsHandler {
-  constructor() {
-    this.cursor = null;
-    this.lookControls = null;
-    this.isMouseDown = false;
-    this.isMouseDownHandledByCursor = false;
+  constructor(cursor, cameraController) {
+    this.cursor = cursor;
+    this.cameraController = cameraController;
+    this.isLeftButtonDown = false;
+    this.isLeftButtonHandledByCursor = false;
     this.isPointerLocked = false;
-    this.dXBuffer = [];
-    this.dYBuffer = [];
 
-    this.registerCursor = this.registerCursor.bind(this);
-    this.registerLookControls = this.registerLookControls.bind(this);
-    this.isReady = this.isReady.bind(this);
     this.addEventListeners = this.addEventListeners.bind(this);
     this.onMouseDown = this.onMouseDown.bind(this);
+    this.onLeftButtonDown = this.onLeftButtonDown.bind(this);
+    this.onRightButtonDown = this.onRightButtonDown.bind(this);
+    this.tearDown = this.tearDown.bind(this);
     this.onMouseMove = this.onMouseMove.bind(this);
     this.onMouseUp = this.onMouseUp.bind(this);
     this.onMouseWheel = this.onMouseWheel.bind(this);
     this.look = this.look.bind(this);
 
-    document.addEventListener("contextmenu", function(e) {
+    this.addEventListeners();
+  }
+
+  tearDown() {
+    document.removeEventListener("mousedown", this.onMouseDown);
+    document.removeEventListener("mousemove", this.onMouseMove);
+    document.removeEventListener("mouseup", this.onMouseUp);
+    document.removeEventListener("wheel", this.onMouseWheel);
+    document.removeEventListener("contextmenu", e => {
```

Comment:
This wont correctly remove the event listener since you are passing a different funciton instance to add/remove. I would just make a static function called preventDefault, and pass that to both.